cmd2.py

```diff
@@ -830,7 +830,7 @@ def __init__(self,
                  command,
                  aliases=(),
                  reformat_prompt="{super_prompt}>> {sub_prompt}",
-                 shared_attributes=None,
+                 shared_attributes={},
```

Comment:
Mutable default arguments are almost always a bad thing in common, they behave in a rather odd manner analogous to static function-scope variables in C:
http://docs.python-guide.org/en/latest/writing/gotchas/

The previous code was attempting to set ``self.shared_attributes`` to an empty dict if the argument was None or to whatever it was supplied as if it isn't None.

Good catch on finding this bug!  But I think a different fix is likely in order.  

What would you think about simply moving the line which sets ``self.shared_attributes`` earlier to prevent attempting to get keys on a None, like so:
```Python
self.shared_attributes = {} if shared_attributes is None else shared_attributes
        if require_predefined_shares:
            for attr in shared_attributes.keys():
                if not hasattr(submenu, attr):
                    raise AttributeError("The shared attribute '{attr}' is not defined in {cmd}. Either define {attr} "
                                         "in {cmd} or set require_predefined_shares=False."
                                         .format(cmd=submenu.__class__.__name__, attr=attr))
```

Reply:
yes, you're right, I'll push the change :+1: